src/DocumentFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Swis\JsonApi\Client;
+
+use Swis\JsonApi\Client\Interfaces\DataInterface;
+use Swis\JsonApi\Client\Interfaces\DocumentInterface;
+use Swis\JsonApi\Client\Interfaces\ItemInterface;
+
+class DocumentFactory
+{
+    /**
+     * @param \Swis\JsonApi\Client\Interfaces\DataInterface $data
+     *
+     * @return \Swis\JsonApi\Client\Interfaces\DocumentInterface
+     */
+    public function make(DataInterface $data): DocumentInterface
+    {
+        $document = new Document();
+        if ($data instanceof ItemInterface) {
+            $document = new ItemDocument();
+        } elseif ($data instanceof Collection) {
+            $document = new CollectionDocument();
+        }
```

Comment:
Wouldn't it be more appropriate to throw an exception here, when trying to _make_ an item of unknown type?